app/src/main/java/no/usn/mob3000/domain/usecase/social/userProfile/FetchFriendsUseCase.kt

```diff
@@ -0,0 +1,28 @@
+package no.usn.mob3000.domain.usecase.social.userProfile
+
+import no.usn.mob3000.data.repository.social.FriendRequestRepository
+import no.usn.mob3000.data.repository.social.FriendsRepository
+import no.usn.mob3000.domain.model.auth.UserProfile
+import no.usn.mob3000.domain.model.social.FriendData
+import no.usn.mob3000.domain.model.social.FriendRequestData
+/**
+ * Fetch.case for friends
+ *
+ * @author 258030
+ * @contributor Husseinabdulameer11
+ * @created: 2024-11-03
+ */
+class FetchFriendsUseCase (
+    private val fetchRepository: FriendsRepository = FriendsRepository(),
+    private val fetchRequestRepository: FriendRequestRepository = FriendRequestRepository()
+){
+    suspend fun fetchFriends(userId: String): Result<List<FriendData>> {
+        return fetchRepository.fetchFriends(userId)
+    }
+    suspend fun fetchNonFriends(userId: String): Result<List<UserProfile>> {
+        return fetchRepository.fetchNonFriends(userId)
+    }
+    suspend fun fetchFriendRequests(): Result<List<FriendRequestData>>{
+        return fetchRequestRepository.fetchFriendRequests()
+    }
```

Comment:
Should be individual use cases. Additionally, spacing.

Reply:
If we used invoke, yes
It's messy to have similar/nearly identical usecases from the same repo in seperate classes, this solution does not make the application work any less inefficent and does not presenet any merge issues.

Reply:
It has to do with how testable it is, multiple functions within one use case means all three need to be mocked and accounted for instead of just one, even if you needed only one of them. And it's harder to override when it they aren't parameters, which can make it difficult to test when the area its being used in is unknown.

Additionally, spacing is still an issue.